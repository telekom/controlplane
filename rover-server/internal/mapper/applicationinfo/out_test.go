// Copyright 2025 Deutsche Telekom IT GmbH
//
// SPDX-License-Identifier: Apache-2.0

package applicationinfo

import (
	"github.com/gkampitakis/go-snaps/snaps"

	. "github.com/onsi/ginkgo/v2"
	. "github.com/onsi/gomega"

	"github.com/telekom/controlplane/rover-server/internal/api"
)

var _ = Describe("ApplicationInfo Mapper", func() {
	Context("FillExposureInfo", func() {
		It("must fill exposure info correctly", func() {
			var applicationInfo = &api.ApplicationInfo{}
			err := FillExposureInfo(ctx, rover, applicationInfo)

			Expect(err).To(BeNil())

			snaps.MatchJSON(GinkgoT(), applicationInfo)
		})

		It("must return an error if the input rover is nil", func() {
			var applicationInfo = &api.ApplicationInfo{}
			err := FillExposureInfo(ctx, nil, applicationInfo)

			Expect(err).ToNot(BeNil())
			Expect(err.Error()).To(ContainSubstring("input rover is nil"))
		})

		It("must return an error if the input application info is nil", func() {
			err := FillExposureInfo(ctx, rover, nil)

			Expect(err).ToNot(BeNil())
			Expect(err.Error()).To(ContainSubstring("input applicationInfo is nil"))
		})
	})

	Context("FillSubscriptionInfo", func() {
		It("must fill subscription info correctly", func() {
			var applicationInfo = &api.ApplicationInfo{}
			err := FillSubscriptionInfo(ctx, rover, applicationInfo)

			Expect(err).To(BeNil())

			snaps.MatchJSON(GinkgoT(), applicationInfo)
		})

		It("must return an error if the input rover is nil", func() {
			var applicationInfo = &api.ApplicationInfo{}
			err := FillSubscriptionInfo(ctx, nil, applicationInfo)

			Expect(err).ToNot(BeNil())
			Expect(err.Error()).To(ContainSubstring("input rover is nil"))
		})

		It("must return an error if the input application info is nil", func() {
			err := FillSubscriptionInfo(ctx, rover, nil)

			Expect(err).ToNot(BeNil())
			Expect(err.Error()).To(ContainSubstring("input applicationInfo is nil"))
		})
	})

	Context("FillApplicationInfo", func() {
		It("must fill application info correctly", func() {
			var applicationInfo = &api.ApplicationInfo{}
			err := FillApplicationInfo(ctx, rover, applicationInfo)

			Expect(err).To(BeNil())

			snaps.MatchJSON(GinkgoT(), applicationInfo)
		})

		It("must return an error if the input rover is nil", func() {
			var applicationInfo = &api.ApplicationInfo{}
			err := FillApplicationInfo(ctx, nil, applicationInfo)

			Expect(err).ToNot(BeNil())
			Expect(err.Error()).To(ContainSubstring("rover resource is not processed and does not contain an application"))
		})

		It("must return an error if the input application info is nil", func() {
			err := FillApplicationInfo(ctx, rover, nil)

			Expect(err).ToNot(BeNil())
			Expect(err.Error()).To(ContainSubstring("input applicationInfo is nil"))
		})
	})

	Context("MapApplicationInfo", func() {
		It("must map application info correctly", func() {
			output, err := MapApplicationInfo(ctx, rover)

			Expect(err).To(BeNil())

			snaps.MatchJSON(GinkgoT(), output)
		})

		It("must return an error if the input rover is nil", func() {
			output, err := MapApplicationInfo(ctx, nil)

			Expect(output).To(BeNil())

			Expect(err).ToNot(BeNil())
			Expect(err.Error()).To(ContainSubstring("input rover is nil"))
		})
	})

})
