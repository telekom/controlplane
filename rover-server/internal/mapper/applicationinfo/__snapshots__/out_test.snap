
[ApplicationInfo Mapper FillExposureInfo must fill exposure info correctly - 1]
{
 "errors": null,
 "exposures": [
  {
   "approval": "Simple",
   "basePath": "/eni/distr/v1",
   "type": "api",
   "upstream": "https://httpbin.org/anything",
   "visibility": "World"
  }
 ],
 "infos": null,
 "irisClientId": "",
 "irisClientSecret": "",
 "irisIssuerUrl": "",
 "irisTokenEndpointUrl": "",
 "name": "",
 "status": "",
 "subscriptions": null,
 "variables": null,
 "warnings": null,
 "zone": ""
}
---

[ApplicationInfo Mapper FillSubscriptionInfo must fill subscription info correctly - 1]
{
 "errors": [
  {
   "cause": "Approval is either rejected or suspended",
   "message": "ApiSubscription 'poc--eni--hyperion/rover-local-sub--eni-distr-v1' is not ready",
   "resource": {
    "apiVersion": "stargate.cp.ei.telekom.de/v1",
    "kind": "ApiSubscription",
    "name": "rover-local-sub--eni-distr-v1",
    "namespace": "poc--eni--hyperion"
   }
  }
 ],
 "exposures": null,
 "infos": null,
 "irisClientId": "",
 "irisClientSecret": "",
 "irisIssuerUrl": "",
 "irisTokenEndpointUrl": "",
 "name": "",
 "status": "blocked",
 "subscriptions": [
  {
   "basePath": "",
   "type": "api"
  }
 ],
 "variables": null,
 "warnings": null,
 "zone": ""
}
---

[ApplicationInfo Mapper FillApplicationInfo must fill application info correctly - 1]
{
 "errors": null,
 "exposures": null,
 "infos": null,
 "irisClientId": "hyperion--rover-local-sub",
 "irisClientSecret": "abcdefghijklmnop",
 "irisIssuerUrl": "https://iris-distcp1-dataplane1.dev.dhei.telekom.de/auth/realms/poc",
 "irisTokenEndpointUrl": "https://iris-distcp1-dataplane1.dev.dhei.telekom.de/auth/realms/poc/protocol/openid-connect/token",
 "name": "",
 "stargateIssuerUrl": "https://stargate-distcp1-dataplane1.dev.dhei.telekom.de:443/auth/realms/poc",
 "stargatePublishEventUrl": "https://stargate-distcp1-dataplane1.dev.dhei.telekom.de/horizon/events/v1",
 "stargateUrl": "https://stargate-distcp1-dataplane1.dev.dhei.telekom.de/",
 "status": "blocked",
 "subscriptions": null,
 "variables": null,
 "warnings": null,
 "zone": ""
}
---

[ApplicationInfo Mapper MapApplicationInfo must map application info correctly - 1]
{
 "errors": [
  {
   "cause": "Approval is either rejected or suspended",
   "message": "ApiSubscription 'poc--eni--hyperion/rover-local-sub--eni-distr-v1' is not ready",
   "resource": {
    "apiVersion": "stargate.cp.ei.telekom.de/v1",
    "kind": "ApiSubscription",
    "name": "rover-local-sub--eni-distr-v1",
    "namespace": "poc--eni--hyperion"
   }
  }
 ],
 "exposures": [
  {
   "approval": "Simple",
   "basePath": "/eni/distr/v1",
   "type": "api",
   "upstream": "https://httpbin.org/anything",
   "visibility": "World"
  }
 ],
 "infos": null,
 "irisClientId": "hyperion--rover-local-sub",
 "irisClientSecret": "abcdefghijklmnop",
 "irisIssuerUrl": "https://iris-distcp1-dataplane1.dev.dhei.telekom.de/auth/realms/poc",
 "irisTokenEndpointUrl": "https://iris-distcp1-dataplane1.dev.dhei.telekom.de/auth/realms/poc/protocol/openid-connect/token",
 "name": "rover-local-sub",
 "stargateIssuerUrl": "https://stargate-distcp1-dataplane1.dev.dhei.telekom.de:443/auth/realms/poc",
 "stargatePublishEventUrl": "https://stargate-distcp1-dataplane1.dev.dhei.telekom.de/horizon/events/v1",
 "stargateUrl": "https://stargate-distcp1-dataplane1.dev.dhei.telekom.de/",
 "status": "blocked",
 "subscriptions": [
  {
   "basePath": "",
   "type": "api"
  }
 ],
 "variables": null,
 "warnings": null,
 "zone": "dataplane1"
}
---
